submissions/AnnaStoyano/oop-exercise/index.js

```diff
@@ -0,0 +1,122 @@
+/* Refer to https://github.com/OleksiyRudenko/a-tiny-JS-world for the task details
+   Complete the below for code reviewers' convenience:
+
+   Code repository: _put repo URL here_
+   Web app: _put project's github pages URL here_
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+
+class Mammal {
+   constructor(species, name, gender, legs, friends, saying) {
+      this.species = species;
+      this.name = name;
+      this.gender = gender;
+      this.legs = legs;
+      this.friends = friends;
+      this.saying = saying;
+   }
+
+   tellAbout(properties) {
+      let propertiesForDispaly = properties || ['species', 'name', 'gender', 'legs', 'friends', 'saying'];
+      return propertiesForDispaly.map(property => `<b>${property}:</b> ${this[property]}`).join('\t');
+   }
+};
+
+class Dog extends Mammal {
+   constructor(name, gender, friends, saying) {
+      super('dog', name, gender, 4, friends, saying || 'guf');
+   }
+
+   tellAbout() {
+      let properties = ['name', 'species', 'friends']; /* properties for display */
+      return super.tellAbout(properties);
+   }
+};
+
+class Cat extends Mammal {
+   constructor(name, gender, friends, saying) {
+      super('cat', name, gender, 4, friends, saying || 'meow');
+   }
+
+   tellAbout() {
+      let properties = ['species', 'name', 'gender', 'legs', 'friends', 'saying']; /* properties for display */
+      return super.tellAbout(properties);
+   }
+
+   get getSaying() {
+      return this.saying;
+   }
+}
+
+class Human extends Mammal {
+   constructor(name, gender, friends, saying) {
+      super('human', name, gender, 2, friends, saying);
+      this.hands = 2;
+      this.saying = saying || `Hello! My name is ${this.name}`;
+   }
+   tellAbout() {
+      let childProperties = arguments[0]; /* array of properties for display from child if it exists*/
+      let properties = ['species', 'name', 'gender', 'legs', 'hands', 'friends', 'saying']; /* properties for display */
+      return super.tellAbout(childProperties || properties);
+   }
+}
+
+class Woman extends Human {
+   constructor(name, friends, saying) {
+      super(name, 'female', friends, saying);
+   }
+   tellAbout() {
+      return super.tellAbout(); /* You can also add array of property to display them */
+   }
+}
+
+class Man extends Human {
+   constructor(name, friends, saying) {
+      super(name, 'male', friends, saying);
+   }
+   tellAbout() {
+      let properties = ['name', 'friends', 'saying'];
+      return super.tellAbout(properties);
+   }
+}
+
+class CatWoman extends Woman {
+   constructor(name, friends) {
+      super(name, friends, 'meow', Cat.prototype.getSaying);
+   }
+   tellAbout() {
+      return super.tellAbout();
+   }
```

Comment:
There is no point to just call an overloaded method from its parent class. All methods from parent classes are inherited by default.